backend/corpora/common/entities/dataset.py

```diff
@@ -79,10 +81,25 @@ def create(
 
         return cls(new_db_object)
 
+    def get_asset(self, asset_uuid) -> typing.Union[Asset, None]:
+        """
+        Retrieve the asset if it exists in the dataset.
+        :param asset_uuid: uuid of the asset to find
+        :return: If the asset is found it is returned, else None is returned.
+        """
+        asset = [asset for asset in self.artifacts if asset.id == asset_uuid]
+        if not asset:
```

Comment:
Can simplify to `return Asset(asset[0]) if asset else None`.

Reply:
Though we might want to handle the case that there is more than 1 asset with the same uuid? That should cause an error that we'd want to investigate.

Reply:
> Though we might want to handle the case that there is more than 1 asset with the same uuid? That should cause an error that we'd want to investigate.

Maintaining unique primary keys is a pretty fundamental database operation. All of the assets are in the same table, therefore I think we can assume there will be no collisions. Do you think this is something we need to test for?

Also I wont be able to write test to verify this error condition, since the DB will prevent me from have duplicate primary keys.